DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_tagged_ports.py

```diff
@@ -0,0 +1,292 @@
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+    ti_to_rx_port_map = get_traffic_port_vlan_mapping(streams, port_map, tg_ports)
+    for row in stats.Rows:
+        if row["Rx Port"] in ti_to_rx_port_map[row["Traffic Item"]]:
+            assert tgen_utils_get_loss(row) == 0.000, \
+                f'No traffic for traffic item : {row["Traffic Item"]} on port {row["Rx Port"]}'
+        else:
+            assert tgen_utils_get_loss(row) == 100.000, \
+                f'Traffic leak for traffic item: {row["Traffic Item"]} on port {row["Rx Port"]}'
+
+
+async def test_vlan_tagged_ports_with_unicast(testbed):
+    """
+    Test Name: Unicast with VLAN tagged ports
+    Test Suite: suite_functional_vlan
+    Test Overview: Test unicast packet forwarding with VLAN tagged ports configured
+    Test Procedure:
+    1. Initiate test params.
+    2. Set links to vlans.
+    3. Map receiving and non receiving dut_ports.
+        Port 1 -> tx_port
+        Port 2 -> rx_port
+    4. Setup packet stream(s) for the unicast packet:
+    5. Send traffic to rx_ports.
+    6. Verify traffic.
+    """
+
+    # 1. Initiate test params.
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgn connections")
+        return
+    device = dent_devices[0].host_name
+    tg_ports = tgen_dev.links_dict[device][0]
+    dut_ports = tgen_dev.links_dict[device][1]
+    traffic_duration = 5
+    tx_port = 0
+
+    await configure_bridge_setup(device, dut_ports)
+
+    # 2. Set links to vlans.
+    await configure_vlan_setup(device, port_map, dut_ports)
+
+    # 3. Map receiving and non receiving dut_ports.
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[2], "ip": "100.1.1.4", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[3], "ip": "100.1.1.5", "gw": "100.1.1.6", "plen": 24, }])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+
+    mac = [f"02:00:00:00:00:0{idx + 1}" for idx in range(4)]
+    streams = {}
+    streams_to_learn = {}
+    for vlan in packet_vids:
+        streams.update({f"VLAN {vlan} from  {mac[src]} to --> {mac[dst]}": {
+                "ip_source": dev_groups[tg_ports[src]][0]["name"],
+                "ip_destination": dev_groups[tg_ports[dst]][0]["name"],
+                "srcMac": mac[src],
+                "dstMac": mac[dst],
+                "type": "raw",
+                "vlan": vlan,
+                "protocol": "802.1Q"
+                } for src, dst in ((0, 1), (0, 2), (0, 3))
+            })
+        for port in port_map:
+            if port["port"] == tx_port or isinstance(port["port"], int) is False:
```

Comment:
Why do you need `isinstance(port["port"], int)`? When can this be False?

Reply:
sorry, had to be `vlan`